### PR TITLE
Added elasticsearch linking support

### PR DESCRIPTION
**Pull request by pazoozooCH**
Hi

I've made a change to your image to support elasticsearch container linking. It's opt-in, so no changes to containers without linking. Would be cool if you could pull this in.

Thanks

### CODE REVIEW

**Comment by pazoozooCH** [comment]
btw, I've pushed a fork to https://registry.hub.docker.com/u/inftec/curator/ in case you want to check it out...

File: curator-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# Automatically set host and port parameters if elasticsearch container is linked with this curator container
+if [[ "$ELASTICSEARCH_PORT_9200_TCP_ADDR" != "" ]]; then
```
Comment (bobrik) [inline curator-entrypoint.sh]: `if [ -z "${ELASTICSEARCH_PORT_9200_TCP_ADDR}" ]; then`

File: curator-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
```
Comment (bobrik) [inline curator-entrypoint.sh:2]: `set -e` is a good idea

File: curator-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+# Automatically set host and port parameters if elasticsearch container is linked with this curator container
+if [[ "$ELASTICSEARCH_PORT_9200_TCP_ADDR" != "" ]]; then
+  set -- --host $ELASTICSEARCH_PORT_9200_TCP_ADDR --port $ELASTICSEARCH_PORT_9200_TCP_PORT $@
+fi
+
+/usr/bin/curator $@
```
Comment (bobrik) [inline curator-entrypoint.sh]: `exec` it

File: Dockerfile [hunk 4]
```diff
@@ -3,4 +3,6 @@ FROM alpine:3.1
 RUN apk --update add python py-pip && \
     pip install elasticsearch-curator==3.0.3
 
-ENTRYPOINT ["/usr/bin/curator"]
+ADD curator-entrypoint.sh /
```
Comment (bobrik) [inline Dockerfile]: can you make it `ADD ./curator-entrypoint.sh /curator-entrypoint.sh`?

**Comment by bobrik** [comment]
I like the idea and would love to merge after you address my comments. Thanks!

**Comment by lbjay** [comment]
:+1: fwiw. this would be very handy.

File: curator-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+set -e
+
+# Automatically set host and port parameters if elasticsearch container is linked with this curator container
+if [ ! -z "$ELASTICSEARCH_PORT_9200_TCP_ADDR" ]; then
```
Comment (bobrik) [inline curator-entrypoint.sh:6]: `! -z` is actually `-n` iirc

**Comment by bobrik** [comment]
Sorry, I missed an email from github with your fixes. Please fix the last comment and squash into single commit. LGTM then.

**Comment by bobrik** [comment]
3.1.0 is on docker hub.

**Pull request closed without merging**